### ISSUE TITLE
linter: `vitest` plugin rules stop triggering when using `test.extend()` to create a custom test function

### ISSUE THREAD

**Issue by michael-wolfenden** [oxc-project/oxc#26675]
### What version of Oxlint are you using?

1.83.0

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
// Replace this with your actual config file contents.
// It can be helpful to keep it minimal, if possible, but please ensure that it actually reproduces the issue you're reporting.
{
  "plugins": ["vitest"],
  "rules": {
    "vitest/no-disabled-tests": "error",
  }
}
```

### Description

Several `vitest` rules (e.g. `no-disabled-tests`) correctly detect violations when using the default `test`/`it` imports from `vitest`, but stop detecting the same violations once a custom test function is created via `test.extend()`. This is a common pattern for injecting fixtures/context into tests, so it seems like a fairly significant gap in coverage for the `vitest` rule set.

### Reproduction

**Triggers `vitest/no-disabled-tests` (expected):**

```tsx
import { test } from "vitest";

test.skip("displays server side errors", async () => {});
```

**Does NOT trigger `vitest/no-disabled-tests` (unexpected):**

```tsx
import { test as baseTest } from "vitest";

interface Context {
  value: boolean;
}

const test = baseTest.extend<Context>({ value: true });

test.skip("displays server side errors", async () => {});
```

### Expected behavior

`test.skip(...)` should still be flagged by `no-disabled-tests` (and presumably other `vitest` rules with similar detection logic) even when `test` is derived from `baseTest.extend(...)` rather than imported directly.

### Actual behavior

The rule only appears to recognize the test function when it's imported directly (or aliased directly) from `vitest`. Once it's reassigned via `.extend()`, the linter seems to lose track of it as a "test" call, so `.skip()` usage goes unflagged.

### Question

Is there a way to configure oxlint so it recognizes a custom test method created via `.extend()` as equivalent to the base `test`/`it`? If not, could support for this be added (e.g. by tracking variables assigned from `<test>.extend()` calls as aliases of the base test function)?
